deflate_slow.c

```diff
@@ -108,11 +108,13 @@ ZLIB_INTERNAL block_state deflate_slow(deflate_state *s, int flush) {
 #else
             {
                 unsigned int mov_fwd = s->prev_length - 2;
-                unsigned int insert_cnt = mov_fwd;
-                if (unlikely(insert_cnt > max_insert - s->strstart))
-                    insert_cnt = max_insert - s->strstart;
+                if (max_insert > s->strstart) {
+                    unsigned int insert_cnt = mov_fwd;
+                    if (unlikely(insert_cnt > max_insert - s->strstart))
+                        insert_cnt = max_insert - s->strstart;
```

Comment:
Note that the above 3 lines are the computation of the `min(mov_fwd, max_insert - s->strstart)`.
This is the same code as what existed before the change.

The only change is the condition under which `insert_string` is called: `if (max_insert > s->strstart)`.